in_toto/slsa_provenance/v0.2/provenance.go

```diff
@@ -0,0 +1,66 @@
+package v02
+
+import "time"
+
+const (
+	// PredicateSLSAProvenance represents a build provenance for an artifact.
+	PredicateSLSAProvenance = "https://slsa.dev/provenance/v0.2"
+)
+
+// ProvenancePredicate is the provenance predicate definition.
+type ProvenancePredicate struct {
+	Builder     ProvenanceBuilder    `json:"builder"`
+	BuildType   string               `json:"buildType"`
+	Invocation  ProvenanceInvocation `json:"invocation"`
```

Comment:
should this use omitempty / a pointer? I see invocation is marked as optional here: https://slsa.dev/provenance/v0.2